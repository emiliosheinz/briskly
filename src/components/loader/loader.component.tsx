import { classNames } from '~/utils/css'

import type { LoaderProps, LoaderVariants } from './loader.types'

export const variantBasedClassName: Record<LoaderVariants, string> = {
  primary: 'bg-primary-50',
  secondary: 'bg-primary-900',
  bad: 'bg-error-700',
}

export function Loader(props: LoaderProps) {
  const { variant = 'secondary' } = props

  return (
    <div data-testid='loader' className='flex gap-1 opacity-90'>
      {Array.from({ length: 3 }).map((_, index) => (
        <span
          key={index}
          className={classNames(
            'inline-flex h-4 w-4 animate-bounce rounded-full',
            variantBasedClassName[variant],
          )}
        />
      ))}
      <p className='sr-only'>Loading...</p>
    </div>
  )
}
