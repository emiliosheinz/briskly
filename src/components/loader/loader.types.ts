export type LoaderVariants = 'primary' | 'secondary' | 'bad'

export type LoaderProps = {
  variant?: LoaderVariants
}
