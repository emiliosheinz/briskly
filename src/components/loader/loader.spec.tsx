import { render, screen } from '~/utils/tests/react-testing-library'

import { Loader, variantBasedClassName } from './loader.component'

describe('Loader Component', () => {
  it('should be correctly rendered', () => {
    const component = render(<Loader />)

    expect(component.asFragment()).toMatchSnapshot()
  })

  it('should apply the secondary variant style as default', () => {
    render(<Loader />)

    const loaderItems = screen.getByTestId('loader').querySelectorAll('span')

    for (const item of loaderItems) {
      expect(item).toHaveClass(variantBasedClassName.secondary)
    }
  })

  it('should apply the primary variant style', () => {
    render(<Loader variant='primary' />)

    const loaderItems = screen.getByTestId('loader').querySelectorAll('span')

    for (const item of loaderItems) {
      expect(item).toHaveClass(variantBasedClassName.primary)
    }
  })

  it('should apply the secondary variant style', () => {
    render(<Loader variant='secondary' />)

    const loaderItems = screen.getByTestId('loader').querySelectorAll('span')

    for (const item of loaderItems) {
      expect(item).toHaveClass(variantBasedClassName.secondary)
    }
  })

  it('should apply the bad variant style', () => {
    render(<Loader variant='bad' />)

    const loaderItems = screen.getByTestId('loader').querySelectorAll('span')

    for (const item of loaderItems) {
      expect(item).toHaveClass(variantBasedClassName.bad)
    }
  })
})
