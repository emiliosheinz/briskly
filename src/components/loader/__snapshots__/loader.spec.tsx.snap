// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should be correctly rendered 1`] = `
<DocumentFragment>
  <div
    class="flex gap-1 opacity-90"
    data-testid="loader"
  >
    <span
      class="inline-flex h-4 w-4 animate-bounce rounded-full bg-primary-900"
    />
    <span
      class="inline-flex h-4 w-4 animate-bounce rounded-full bg-primary-900"
    />
    <span
      class="inline-flex h-4 w-4 animate-bounce rounded-full bg-primary-900"
    />
    <p
      class="sr-only"
    >
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
